ckanext/showcase/commands/migrate.py

```diff
@@ -120,3 +138,12 @@ def _find_duplicates(self, lst):
         ['two', 'three']
         '''
         return list(set(x for x in lst if lst.count(x) >= 2))
+
+    def _gen_new_title(self, title, related_id):
+        name = munge_title_to_name(title)
+        pkg_obj = model.Session.query(model.Package).filter_by(name=name).first()
+        if pkg_obj:
+            title.replace('duplicate_', '')
+            return 'duplicate_' + title + '_' + self._get_related_dataset(related_id) + '_' + str(uuid.uuid4())[:3]
```

Comment:
Why get the related dataset here? Why not just use the `related_id`. It's unique and also means you don't have to use the uuid. If we have several related items associated with one dataset there's still a (remote) chance of a duplicate title here.